### ISSUE TITLE
Chef 13 Deprecation warnings

### ISSUE THREAD

**Issue by spuder** [NetDocuments/ad-join-cookbook#20]
Getting these warning on chef 12.19.33

```
10.254.60.182 [2017-02-22T19:47:03+00:00] INFO: Processing domain_join[default] action join (nd-ad-join::default line 31)
10.254.60.182 [2017-02-22T19:47:03+00:00] WARN: Cloning resource attributes for windows_task[chef ad-join] from prior resource
10.254.60.182 Previous windows_task[chef ad-join]: c:/chef/cache/cookbooks/ad-join/resources/domain_join.rb:58:in `block in class_from_file'
10.254.60.182 Current  windows_task[chef ad-join]: c:/chef/cache/cookbooks/ad-join/resources/domain_join.rb:113:in `block in class_from_file' (CHEF-3694)c:/chef/cache/cookbooks/ad-join/resources/domain_join.rb:113:in `block in class_from_file'
```

**Comment by spuder** [comment]
13.4.19 has fixed many of the problems https://github.com/chef/chef/blob/master/CHANGELOG.md#changes-since-13419-release

**Comment by spuder** [comment]
Fixed in #29
